### ISSUE TITLE
How to change the default python set-up the TLJH uses

### ISSUE THREAD

**Issue by ThorAlexis** [jupyterhub/the-littlest-jupyterhub#420]
Hi!

Is there a way to change the default python that TLJH uses?

According to (http://tljh.jupyter.org/en/latest/topic/installer-actions.html), TLJH is run from a Python 3 virtual environment located in /opt/tljh/hub. Is there a way to change this such that TLJH uses a Python located in another directory?

I'm trying to set-up TLJH for running IDAES (https://idaes-pse.readthedocs.io/en/latest/index.html).

Installation of IDAES gives me a Conda environment set-up. 
![2](https://user-images.githubusercontent.com/53846466/63329365-c164be00-c36c-11e9-8267-7f659c6151f5.JPG)

I already added the Python IDAES set-up to my $PATH environment variables so I could already run IDAES using the Terminal and SSH.
![3](https://user-images.githubusercontent.com/53846466/63329788-8a42dc80-c36d-11e9-9ccc-e16e2beead80.JPG)

However, when I open an .ipynb file in TLJH directly and run it, it fails, I guess since it uses the Python in  /opt/tljh/hub, which is not set up to run IDAES.

Quite new to this. Any suggestions would be highly appreciated. Thanks in advance!

![4](https://user-images.githubusercontent.com/53846466/63329992-ed347380-c36d-11e9-9d13-34f0fd76b7e8.JPG)

**Comment by ThorAlexis** [comment]
oops, nevermind. figured out a solution.

**Comment by raghukamath** [comment]
Hi @ThorAlexis , can you please share the solution to this problem? Cheers

**Comment by TobiGiese** [comment]
Please release your solution. I am getting desperate with the same issue right now... thanks!

**Comment by ThorAlexis** [comment]
Hi, 
It's been a while and I haven't used it since so I can't remember well, but it think I only added "sudo" to each command in the installation procedure. I've checked the IDEAS page today and it seems that the installation instructions have already changed so I'm not sure if this will work. Anyway, below is what I did before:

---------------------------------------------
After you install TLJH

(a) Linux installation of IDAES 

Open terminal and enter: sudo -E apt-get install gcc g++ make libboost-dev

(b) Obtain the source code for IDAES from GitHub:
sudo -E git clone https://github.com/IDAES/idaes-pse.git

(c) Download and compile the AMPL Solver Library (ASL) and compile external property functions; this is required for steam properties and cubic equations of state. This step is optional but highly recommended.

sudo -E mkdir ASL
cd ASL/
sudo -E wget https://ampl.com/netlib/ampl/solvers.tgz
sudo -E tar -xf solvers.tgz
cd solvers/
sudo -E ./configure
sudo -E make
export ASL_BUILD=`pwd`/sys.`uname -m`.`uname -s`
cd /home/jupyter-**user**/idaes-pse/ 
**(user refers to your username you chose. E.g. if my username is fuku cd /home/jupyter-fuku/idaes-pse/)**
sudo -E make

(d) Install the required Python packages:

sudo -E pip install -r requirements.txt
sudo -E python setup.py develop  # or "install"

(8) Install ipopt

sudo -E conda install -c conda-forge ipopt
